neptune_tf/sync/tensorflow_data_sync.py

```diff
@@ -33,36 +34,25 @@ def __init__(self, project, path):
         self._project = project
         self._path = path
 
-    @staticmethod
-    def requirements_installed():
-        # pylint:disable=unused-variable,unused-import
-        try:
-            import tensorflow
-            return True
-        except ImportError:
-            return False
-
     def run(self):
-        import tensorflow as tf
         for root, _, run_files in os.walk(self._path):
             for run_file in run_files:
                 try:
-                    self._load_single_run(os.path.join(root, run_file), tf)
+                    self._load_single_run(os.path.join(root, run_file))
                 except Exception as e:
                     print("Cannot load run from file '{}'. ".format(run_file) + str(e), file=sys.stderr)
                     traceback.print_exc(e)
 
-    def _load_single_run(self, path, tf):
+    def _load_single_run(self, path):
         click.echo("Loading {}...".format(path))
         run_path = os.path.relpath(path, self._path)
-        run_name = re.sub(r'[^0-9A-Za-z_\-]', '_', run_path)
-        if not self._experiment_exists(run_name):
-            with self._project.create_experiment(name=run_name,
+        run_id = re.sub(r'[^0-9A-Za-z_\-]', '_', run_path)
+        if not self._experiment_exists(run_id, run_path):
+            with self._project.create_experiment(name=run_id,
```

Comment:
Ale po co używać tego jako name'a? Weźmy patha tak jak mówiłem. Jedyny cel istnienia tego `run_id` to, że tagi nie obsługują pewnych znaków. `name` je obsługuje. Poza tym jak sprawdzasz czy eksperyment istnieje to porównujesz z `run_path` więc to bug.